Sources/Stream.swift

```diff
@@ -1,18 +1,22 @@
 public protocol Sending: AsyncSending {
-    func send(_ data: Data, timingOut deadline: Double) throws
+    func send(_ data: Sent, timingOut deadline: Double) throws
     func flush(timingOut deadline: Double) throws
 }
 
 public protocol Receiving: AsyncReceiving {
-    func receive(upTo byteCount: Int, timingOut deadline: Double) throws -> Data
+    func receive(upTo byteCount: Int, timingOut deadline: Double) throws -> Received
 }
 
 public protocol SendingStream: Closable, Sending {}
 public protocol ReceivingStream: Closable, Receiving {}
-public protocol Stream: SendingStream, ReceivingStream {}
+public protocol Stream: SendingStream, ReceivingStream {
+    associatedtype Interchanged
```

Comment:
Is `Interchanged` a good name here? I'm not really sure what it's trying to describe.